Modules/socketmodule.c

```diff
@@ -1724,8 +1724,10 @@ getsockaddrarg(PySocketSockObject *s, PyObject *args,
 
         struct sockaddr_un* addr = &addrbuf->un;
 #ifdef __linux__
-        if (path.len > 0 && *(const char *)path.buf == 0) {
-            /* Linux abstract namespace extension */
+        if (path.len == 0 || (path.len > 0 && *(const char *)path.buf == 0)) {
```

Comment:
It can be simpler (len is always >= 0):
```suggestion
        if (path.len == 0 || *(const char *)path.buf == 0) {
```